forum/models/model_utils.py

```diff
@@ -885,3 +890,339 @@ def get_threads(
+                "subscribed_commentable_ids": [],
+                "subscribed_user_ids": [],
+                "follower_ids": [],
+                "id": user["external_id"],
+                "upvoted_ids": upvoted_ids,
+                "downvoted_ids": downvoted_ids,
+                "default_sort_key": user["default_sort_key"],
+            }
+        )
+
+    if params.get("course_id"):
+        threads = thread_model.find(
+            {
+                "author_id": user["external_id"],
+                "course_id": params["course_id"],
+                "anonymous": False,
+                "anonymouse_to_peers": False,
+            }
+        )
+        comments = comment_model.find(
+            {
+                "author_id": user["external_id"],
+                "course_id": params["course_id"],
+                "anonymous": False,
+                "anonymouse_to_peers": False,
+            }
+        )
+        if params.get("group_ids"):
+            specified_groups_or_global = params["group_ids"] + [None]
+            group_query = {
+                "_id": {"$in": [thread["_id"] for thread in threads]},
+                "$and": [
+                    {"group_id": {"$in": specified_groups_or_global}},
+                    {"group_id": {"$exists": False}},
+                ],
+            }
+            group_threads = CommentThread().find(group_query)
+            group_thread_ids = [str(thread["_id"]) for thread in group_threads]
+            threads_count = len(group_thread_ids)
+            filtered_comments = []
+            for comment in comments:
+                if not comment["context"] == "standalone":
+                    filtered_comments.append(comment)
+            comment_thread_ids = [
+                str([comment["comment_thread_id"]]) for comment in filtered_comments
+            ]
+            group_query = {
+                "_id": {"$in": [ObjectId(tid) for tid in comment_thread_ids]},
+                "$and": [
+                    {"group_id": {"$in": specified_groups_or_global}},
+                    {"group_id": {"$exists": False}},
+                ],
+            }
+            group_comment_threads = thread_model.find(group_query)
+            group_comment_thread_ids = [
+                str(thread["_id"]) for thread in group_comment_threads
+            ]
+            comments_count = sum(
+                1
+                for comment_thread_id in comment_thread_ids
+                if comment_thread_id in group_comment_thread_ids
+            )
+        else:
+            thread_ids = [str(thread["_id"]) for thread in threads]
+            threads_count = len(thread_ids)
+            filtered_comments = []
+            for comment in comments:
+                if not comment["context"] == "standalone":
+                    filtered_comments.append(comment)
+            comment_thread_ids = [
+                str([comment["comment_thread_id"]]) for comment in filtered_comments
+            ]
```

Comment:
we can move this portion outside if-else statement, as it is identical in both blocks.